### ISSUE TITLE
`--with-tag-message` has no effect with `--tag`

### ISSUE THREAD

**Issue by muja** [orhun/git-cliff#836]
Note: Title changed "multiple repositories" to "`--tag`"

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the bug

When using `--tag foo --with-tag-message bar` with multiple repositories, the `message` is `null` and the `version` is `foo` for all repositories.

### Steps To Reproduce

```
git cliff --repository repo1 repo2 --tag 0.1.0 --with-tag-message "test release" --context
```

### Expected behavior

`message` is `bar`

### Software information

- Project version: 2.4.0

**Comment by muja** [comment]
I realized this has nothing to do with multiple repositories. It has to do with the `--tag <VERSION>` flag. It doesn't work for single repositories either.

Works:

`git cliff --bump --with-tag-message foo`

Doesn't work:


`git cliff --tag 1.2.3 --with-tag-message foo`

**Comment by orhun** [comment]
Hello, thanks for reporting 👋🏼 

> I realized this has nothing to do with multiple repositories. It has to do with the --tag <VERSION> flag. It doesn't work for single repositories either.

> Doesn't work:

What are you expecting the `--tag` flag to do here exactly? It is just for setting the tag for the unreleased changes (i.e. it does not actually create the tag)

Also, can you give more information about your Git setup and the changelog format you want to achieve at the end? 🐻

**Comment by muja** [comment]
--tag works, it sets the tag for unreleased changes correctly, but the message is empty even though `--with-tag-message` is provided.

Our commands for releasing look like this:

```bash
VERSION=$(git cliff --bumped-version)
MESSAGE="This release fixes the year old bug #1337"
git cliff --output --tag "$VERSION" --with-tag-message "$MESSAGE"
git add .
git commit -F- <<EOF
v$VERSION

$MESSAGE
EOF
git tag $VERSION -m "$MESSAGE"
git push
git push --tags
```

However using `git cliff --tag $VERSION` instead of `git cliff --bump` unsets the message passed with `--with-tag-message`.

The expected behavior would be that the message is set in either case.

If something is wrong with our process or you have an idea for improving it, I'm all ears :)

**Comment by orhun** [comment]
Thanks for the detailed explanation, now I see the issue :)

It turned out to be a bug and it should be fixed in #854 - can you possibly build from source and test it? 🐻

**Comment by muja** [comment]
Done, it works with that fix. Thank you, really appreciate it
